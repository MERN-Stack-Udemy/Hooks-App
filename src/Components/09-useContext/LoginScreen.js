import React, { useContext } from 'react'
import { useForm } from '../../Hooks/useForm';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

  const [ {userName}, handleChange ] = useForm({});

  const {setUser} = useContext(UserContext);

  const handleSubmit = (e) =>{
    e.preventDefault();
    setUser({
      id: new Date().getTime(),
      name: userName
    })
  }

  return (
    <div>
      <h1>Login Screen</h1>
      <hr/>
      <div className="container-md">
        <form onSubmit={ handleSubmit }>
          <input
            autoComplete="off"
            className="form-control"
            name="userName"
            onChange={handleChange}
            placeholder="User name"
            type="text"
            value={userName}
          />

          <button 
            className="btn btn-outline-primary mt-2 btn-block" 
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  )
}
