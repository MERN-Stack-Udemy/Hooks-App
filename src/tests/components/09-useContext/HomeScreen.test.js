import { mount } from "enzyme"
import { HomeScreen } from "../../../Components/09-useContext/HomeScreen"
import { UserContext } from "../../../Components/09-useContext/UserContext";

describe('test on <HomeScreen/>', () => {
    
    const user = {
        name: 'smog',
        email: 'smog@gmail.com'
    }
    const wrapper = mount(
        <UserContext.Provider value = {{user}}> 
            <HomeScreen/>
        </UserContext.Provider>
    );
    
    test('should be display to match with the snapshot', () => {
        expect( wrapper ).toMatchSnapshot();    
    })
    
})
