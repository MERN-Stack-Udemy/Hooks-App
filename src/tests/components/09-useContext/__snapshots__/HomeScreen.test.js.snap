// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <HomeScreen/> should be display to match with the snapshot 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre
    className="container"
  >
    {
  "name": "smog",
  "email": "smog@gmail.com"
}
  </pre>
</div>
`;
